daemon/algod/api/server/v2/handlers.go

```diff
@@ -943,6 +950,221 @@ func (v2 *Handlers) TealDryrun(ctx echo.Context) error {
 	return ctx.JSON(http.StatusOK, response)
 }
 
+// UnsetSyncRound removes the sync round restriction from the ledger.
+// (DELETE /v2/ledger/sync)
+func (v2 *Handlers) UnsetSyncRound(ctx echo.Context) error {
+	err := v2.Node.UnsetSyncRound()
+	if err != nil {
+		return internalError(ctx, err, errFailedSettingSyncRound, v2.Log)
+	}
+	return ctx.NoContent(http.StatusOK)
+}
+
+// SetSyncRound sets the sync round on the ledger.
+// (POST /v2/ledger/sync/{round})
+func (v2 *Handlers) SetSyncRound(ctx echo.Context, round uint64) error {
+	err := v2.Node.SetSyncRound(round)
+	if err != nil {
+		switch err {
+		case catchup.ErrSyncRoundInvalid:
+			return badRequest(ctx, err, errFailedSettingSyncRound, v2.Log)
+		default:
+			return internalError(ctx, err, errFailedSettingSyncRound, v2.Log)
+		}
+	}
+	return ctx.NoContent(http.StatusOK)
+}
+
+// GetSyncRound gets the sync round from the ledger.
+// (GET /v2/ledger/sync)
+func (v2 *Handlers) GetSyncRound(ctx echo.Context) error {
+	set, rnd, err := v2.Node.GetSyncRound()
+	if err != nil {
+		return internalError(ctx, err, errFailedRetrievingSyncRound, v2.Log)
+	}
+	if !set {
+		return notFound(ctx, fmt.Errorf("sync round is not set"), errFailedRetrievingSyncRound, v2.Log)
+	}
+	return ctx.JSON(http.StatusOK, generated.GetSyncRoundResponse{Round: rnd})
+}
+
+// GetRoundDeltas returns the deltas for a given round.
+// This should be a combination of AccountDeltas, KVStore Deltas, etc.
+// (GET /v2/deltas/{round})
+func (v2 *Handlers) GetRoundDeltas(ctx echo.Context, round uint64) error {
+	ads, err := v2.Node.LedgerForAPI().GetAccountDeltasForRound(basics.Round(round))
+	if err != nil {
+		return internalError(ctx, err, errFailedRetrievingAccountDeltas, v2.Log)
+	}
+	kvds, err := v2.Node.LedgerForAPI().GetKvDeltasForRound(basics.Round(round))
+	if err != nil {
+		return internalError(ctx, err, errFailedRetrievingKvDeltas, v2.Log)
+	}
+
+	var accts []generated.AccountBalanceRecord
+	var apps []generated.AppResourceRecord
+	var assets []generated.AssetResourceRecord
+	var keyValues []generated.KvDelta
+
+	consensusParams, err := v2.Node.LedgerForAPI().ConsensusParams(basics.Round(round))
+	if err != nil {
+		return internalError(ctx, fmt.Errorf("unable to retrieve consensus params for round %d", round), errInternalFailure, v2.Log)
+	}
+	hdr, err := v2.Node.LedgerForAPI().BlockHdr(basics.Round(round))
+	if err != nil {
+		return internalError(ctx, fmt.Errorf("unable to retrieve block header for round %d", round), errInternalFailure, v2.Log)
+	}
+
+	for key, kvDelta := range kvds {
+		var keyBytes = []byte(key)
+		keyValues = append(keyValues, generated.KvDelta{
+			Key:       &keyBytes,
+			PrevValue: &kvDelta.OldData,
+			Value:     &kvDelta.Data,
+		})
+	}
+
+	for _, record := range ads.GetAllAccounts() {
```

Comment:
@Eric-Warehime Double checking:  Is there a reason the highlighted logic builds  `generated.Account` inline rather than using `AccountDataToAccount`?  If no, can it be switched?

* The topic caught my attention because I've recently encountered bugs stemming from partially initialized structs.  
  * https://github.com/algorand/go-algorand/pull/4669 shows one example.
  * Related - https://github.com/algorand/go-algorand/issues/4761 makes an analogous request to avoid partial struct initialization.
* For example, the highlighted logic does _not_ populate `TotalBoxes`.  Presumably, it's accidentally _not_ populated.

Reply:
@Eric-Warehime I opened a child PR to address the incomplete `models.Account` initialization.  Can you review https://github.com/Eric-Warehime/go-algorand/pull/2 to see if you're comfortable pulling it in?

I do not have sufficient privileges to request review on your fork nor run tests.